net/ip.cpp

```diff
@@ -0,0 +1,86 @@
+// SPDX-FileCopyrightText: 2023 David Zero <zero-one@zer0-one.net>
+//
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <cmath>
+#include <cstdint>
+#include <cstring>
+#include <span>
+
+#include <fmt/core.h>
+
+#include "ip.h"
+
+namespace net {
+
+std::string ipv4_serialize(std::uint32_t addr) {
+    std::string out = "";
+    std::uint32_t n = addr;
+
+    for (std::size_t i = 1; i < 5; i++) {
+        out.insert(0, std::to_string(n % 256));
+
+        if (i != 4) {
+            out.insert(0, ".");
+        }
+
+        n = (std::uint32_t)floor(n / 256.0);
```

Comment:
Is the u32 -> double -> floor -> u32 dance necessary? Why not just something like `n /= 256`?

Reply:
n >> 8